### ISSUE TITLE
[Accessibility]: Search notebooks management ARC checklist issues

### ISSUE THREAD

**Issue by rvantonder** [sourcegraph/sourcegraph#35167]
Notebooks Page 10a:

Headings:
- [ ] The h1 heading on this page needs to be h2

Images:
- Two icons on Notebook page
  - [ ] Notebook icon SVG needs `img` on SVG and aria labelby
<img width="419" alt="Screen Shot 2022-05-09 at 16 07 09" src="https://user-images.githubusercontent.com/12631702/167492938-9e1b7a76-b76b-4c70-a2c5-515ca15d8d8d.png">

  - [ ] star icon in the list items
<img width="432" alt="Screen Shot 2022-05-09 at 16 34 32" src="https://user-images.githubusercontent.com/12631702/167493114-d07234ed-4892-417a-846e-8d3a9128c217.png">

Forms:
- [ ] 'placeholder used for label' 
<img width="429" alt="Screen Shot 2022-05-09 at 16 11 18" src="https://user-images.githubusercontent.com/12631702/167493204-c1b4677b-2412-4379-877d-91de61bf79d8.png">

--- 

Notebooks Editing 10b:

Headings:
- [ ] The h1 heading on this page needs to be h2 (fixed by above)

Images:
- Allo Icons on edit menu on the right needs
  - [ ] `img` on SVG and aria labelby

Links:
- [ ] Open in new tab needs `add text to the link` (match the button link text)

Button:
- [ ] Edit buttons need accessibility text (match the button link text)
<img width="436" alt="Screen Shot 2022-05-09 at 16 35 51" src="https://user-images.githubusercontent.com/12631702/167493330-e71f60f9-01d9-441e-af9f-fbfaf80cb22e.png">

Forms:
- [ ] 'placeholder used for label' needs fixing in the edit input
<img width="434" alt="Screen Shot 2022-05-09 at 16 37 23" src="https://user-images.githubusercontent.com/12631702/167493533-b10659fb-00dd-45b1-8cb3-4029362976af.png">

**Comment by gitstart-sourcegraph** [comment]
Hello @rvantonder
With regards to this requirement `placeholder used for label` 
Want to clarify that this requirement entails adding `label` to this `input` element? 
If yes, what should be the label's value / orientation(top, left, right)?
BTW we have other similar use cases on other pages ([e.g](http://localhost:3080/contexts?visible=9)) where `FilteredConnection` is used, does this applies on all those use cases as well?
CC: @umpox @limitedmage

**Comment by umpox** [comment]
@gitstart-sourcegraph 

For the first question, this is only adding an `aria-label` as `placeholder` is not accessible for screen readers. The text should match the `placeholder` in this case. As for FilteredConnection, yes that is a good point - we can automatically do this, I made a PR here to fix this: https://github.com/sourcegraph/sourcegraph/pull/36643

**Comment by limitedmage** [comment]
@rvantonder Can you clarify #2 and #3?

I *think* #2 means that the "open in a new tab" icons on ["Getting Started" page](https://sourcegraph.com/notebooks?tab=getting-started) need labels. 
<details>

![CleanShot 2022-06-06 at 11 39 34](https://user-images.githubusercontent.com/206864/172225285-18b6cfe5-8eeb-403c-af05-e04ce8807fa2.png)
</details>

**Comment by rvantonder** [comment]
2 refers to this element "Open in new tab" element, created by certain kinds of results (search results) in notebooks:

![Screen Shot 2022-06-10 at 3 15 40 PM](https://user-images.githubusercontent.com/888624/173157701-a1cb3424-c12c-4442-99ba-a043aa514e4c.png)

3 refers to the side panel button palette. I think the screenshot here was an accident--I paired with someone else on this issue and didn't take the screenshot myself, this should clear it up:


![Screen Shot 2022-06-10 at 3 16 19 PM](https://user-images.githubusercontent.com/888624/173157850-c0c44bb0-ebff-4540-954c-baafc581f694.png)
